scripts/bittensor-recovery-script/main.ts

```diff
@@ -0,0 +1,133 @@
+import { ApiPromise, WsProvider, Keyring } from '@polkadot/api';
+import readlineSync from 'readline-sync';
+import { BN } from 'bn.js';
+import { cryptoWaitReady } from '@polkadot/util-crypto';
+import * as ed from '@noble/ed25519';
+import { sha512 } from '@noble/hashes/sha512';
+
+// Constants
+const DECIMALS = 9;
+const PLANCK = new BN(10).pow(new BN(DECIMALS)); // 1 unit = 10^9 Planck
+const SS58_FORMAT = 42; // SS58 address format for the Bittensor network
+
+// Assign SHA512 hash function for noble-ed25519 compatibility
+ed.etc.sha512Sync = (...m) => sha512(ed.etc.concatBytes(...m));
+
+/**
+ * Transfers funds on the Bittensor network.
+ * @param privateKey - The raw Ed25519 private key in hex format.
+ * @param destination - The SS58 destination address.
+ * @param amount - The amount to transfer in Planck units.
+ * @param endpoint - The WebSocket endpoint for the Bittensor network.
+ */
+async function transferFunds(privateKey: string, destination: string, amount: string, endpoint: string) {
+  console.log("Initializing cryptography...");
+  await cryptoWaitReady();
+
+  console.log("Connecting to the Bittensor network...");
+  const provider = new WsProvider(endpoint);
+  const api = await ApiPromise.create({ provider });
+
+  console.log("Creating keyring...");
+  const keyring = new Keyring({ type: 'ed25519', ss58Format: SS58_FORMAT });
+
+  // Prepare the private key
+  let privateKeyBytes = Buffer.from(privateKey, 'hex');
+
+  // Derive the public key from the private key
+  const derivedPublicKeyBytes = await ed.getPublicKey(privateKeyBytes);
```

Comment:
@notatestuser this code doesnt seem to derive the same public key as I'd expect it to. James's vault (`m9v62s7b7114nqs7tf421mx0`) has the address `5DDRNSii3jUdb4yWot6ZvzeXg3DtUG7vWXQzVJkbVnZKYqUG` and when I run the script using his EdDSA/Ed25519 private key from the DR tool (can share via slack) im getting a different address. Signature seems to be accepted ok though....

My question is are there any obvious mess ups here? Perhaps im missing out some manipulations that TSS performs?

Reply:
I've been down the rabbit hole of deriving the keys without 'clamping', and also ensuring the correct padding on the privateKey buffer. They did derive different public addresses but not the address I was expecting

Reply:
Two ways to do it.

1. Ask them to input the public key from the DR tool https://github.com/IoFinnet/io-vault-disaster-recovery-cli/blob/master/scripts/xrpl-tool/main.ts#L37
2. Derive it yourself in the script using this way https://github.com/IoFinnet/io-vault-disaster-recovery-cli/blob/master/scripts/xrpl-tool/main.ts#L207

Reply:
Want me to give that a try @JacobPlaster or are you on it?

Reply:
Well I did try accepting the public key as an input, and that did indeed derive the correct expected address. But then i'd get invalid signature.

I'd be very greatful if you could take a whack at it, I've sunk quite a bit of time into it going down the chatGPT rabbit hole to remove clamping, adding padding, different crypto libraries ect...

Reply:
Do you have that version where you loaded the key by input? I would imagine it was just using the address in a wrong way

Reply:
If not, I can work with this

Reply:
Try that with the latest commit @JacobPlaster